Birth & Citizenship Information

FabrizioMoggio opened the pull request:
Adding place of birth and nationality (the one related to the ID Document).

#### What type of PR is this?
* enhancement/feature

#### What this PR does / why we need it:
Those parameters are required or strongly recommended by various international and national regulations 

#### Which issue(s) this PR fixes:
https://github.com/camaraproject/KnowYourCustomer/issues/211

FabrizioMoggio committed:
Birth & Citizenship Information

https://github.com/camaraproject/KnowYourCustomer/discussions/194

GillesInnov35 commented:
Thanks @FabrizioMoggio, I think nationalityScore should not be calculated nor returned. Same decision have been made for country for example.
BR
Gilles

FabrizioMoggio commented:
> Thanks @FabrizioMoggio, I think nationalityScore should not be calculated nor returned. Same decision have been made for country for example. BR Gilles

you are totally right, it is a code not a free text so the score doesn't make sense.

FabrizioMoggio committed:
Nationality output

https://github.com/camaraproject/KnowYourCustomer/pull/204#issuecomment-2875276864

FabrizioMoggio commented:
is there any other comment on this PR? The received comments have been addressed.

HuubAppelboom commented:
@FabrizioMoggio The PR seems fine to me.

GillesInnov35 commented:
Helllo @FabrizioMoggio , I think we can include those updates if ok for all.
Thanks

ToshiWakayama-KDDI commented:
Hi @FabrizioMoggio , @GillesInnov35 , @HuubAppelboom ,

Thanks for the PR.  I am checking it internally with our team now.  Sorry for the delay.

Thanks,
Toshi

GillesInnov35 commented:
hello, same PR should be created for kyc-fill-in when migrated in the new repo.

ToshiWakayama-KDDI commented:
Hi @FabrizioMoggio , @HuubAppelboom , @GillesInnov35 

The PR seems fine to me.  Sorry for the slow response.

BR
Toshi

GillesInnov35 commented:
thanks @FabrizioMoggio , I pointed some corrections to be made.
- description of **_nationality_** `ID documenent provided`
- response example may include **_placeOfBirthMatch_** (and perhaps score) and **_nationalityMatch_** as they are in the example of request body
- remove `'` which has been added at the end of _genderMatch_ description 

may I let you modify your proposal.
Thanks a lot
Gilles

FabrizioMoggio committed:
Updates according to comments

https://github.com/camaraproject/KnowYourCustomer/pull/204#issuecomment-2903639272

GillesInnov35 reviewed (dismissed):
LGTM  thanks

FabrizioMoggio commented:
is there something missing to merge it?

fernandopradocabrillo commented on code/API_definitions/kyc-match.yaml:
```diff
@@ -325,7 +332,14 @@ components:
+            
+        placeOfBirth:
+          type: string
+          description: City and country where the customer was born.
```
what would be the formato to input this information? is it valid if it is only the country or the city, or it needs to be both? is the info separated by any separator?

FabrizioMoggio replied:
It should only be the City, the Country already is indeed in the variable "nationality". I will modify the text as:

 description: name of the City where the customer was born.

FabrizioMoggio replied:
And I must also remove "nationality" that is a duplicate of "country"

FabrizioMoggio committed:
Update code/API_definitions/kyc-match.yaml

Co-authored-by: Fernando Prado Cabrillo <pradocabrillo.fernando@gmail.com>

FabrizioMoggio committed:
cityOfBirth

https://github.com/camaraproject/KnowYourCustomer/pull/204#discussion_r2137259189

FabrizioMoggio committed:
countryAtBirth

https://github.com/camaraproject/KnowYourCustomer/discussions/194#discussioncomment-13442947

FabrizioMoggio committed:
TMF Alignement

https://github.com/camaraproject/KnowYourCustomer/discussions/194#discussioncomment-13443360

FabrizioMoggio committed:
Test for Birth & Citizenship Information

https://github.com/camaraproject/KnowYourCustomer/issues/211

FabrizioMoggio commented:
@ToshiWakayama-KDDI can I move my PRs into the new KYV Match repository?

ToshiWakayama-KDDI commented:
Hi @FabrizioMoggio ,

Yes, please.  I have to ask you to create a new PR in the new kyc-match repo, because it is not possible to move a PR to a different repo, I have heard.  Sorry about that.  I think creating the new PR only should be fine.  If needed, you can note the issue #211 and this PR PR #204 as the references.

Many thanks,
Toshi

ToshiWakayama-KDDI commented:
Hi @FabrizioMoggio ,

In addition, I think we need the same attributes/parameters for kyc-fill-in api.  Is it ok for me to do that?  Or, would you like to do it?

Thanks,
Toshi

FabrizioMoggio commented:
@ToshiWakayama-KDDI I just notice that the new Repo doesn't have the Linting tools yet. Would you like me to add them?

FabrizioMoggio commented:
https://github.com/camaraproject/KnowYourCustomerMatch/issues/6
